builtInFunctions/esdtDeleteMetadata.go

```diff
@@ -49,12 +49,12 @@ func NewESDTDeleteMetadataFunc(
 	}
 
 	e := &esdtDeleteMetaData{
-		keyPrefix:      []byte(baseESDTKeyPrefix),
-		marshaller:     args.Marshalizer,
-		funcGasCost:    args.FuncGasCost,
-		accounts:       args.Accounts,
-		allowedAddress: args.AllowedAddress,
-		delete:         args.Delete,
+		keyPrefix:        []byte(baseESDTKeyPrefix),
+		marshaller:       args.Marshalizer,
+		funcGasCost:      args.FuncGasCost,
+		accounts:         args.Accounts,
+		allowedAddresses: args.AllowedAddresses,
```

Comment:
when creating this component you should choose only the address which is in this shard. No need to keep the whole list. It would make faster the isAllowed function as well. You could check that there is one AllowedAddress per shard and keep the one for your own shard. As for metachain - you could take the first one. Also make sure that there is number of shards addresses in the list.

Reply:
Then, we should not have a new implementation on this vm-common as the address filtering can be easily be done only in elrond-go. What do you think, @sasurobert ?

Reply:
Maybe - when we create the built in functions - so we do not need this PR at this level - only in elrond-go.